V1.1.0 Enable dynamic clear of expired records and add csl service (#32)
UKHomeOffice/hof-rds-api: README.md

With the server running you can run the main app with save and return lookup UI functionality.
See details of how to do this in [modern slavery](https://github.com/UKHomeOffice/modern-slavery) application

## Scheduled jobs

hof-rds-api has funtionality to remove expired records from database tables based on configuration set per service in its `db_tables_config.json`.

Values can be set per table to configure deletion jobs. A single job can be added per table in its configuring object, and if required, additional jobs can be added as an array `customCronJobs`.

### Deletion job configuration

Required values must be configured per job. Optional values can be set or left to default.

`tableName`: The name of the table to run the job for. If using `customCronJobs` set this individually for each custom job. Required value.
`dataRetentionPeriodType`: Calculate which records to retain based period type - optionally "calendar" or "business". Defaults to "calendar".

> `dataRetentionInDays`: The number of days of the chosen period back from the current date to retain records from. Required value.

Reviewer: ```suggestion
`dataRetentionInDays`: The number of days of the chosen period for which data should be retained. Required value.
```

Author: I think its important to state that the retention period is measured backwards from the current date. How about: `The number of days of the chosen period for which data should be retained, calculated backwards from the current date and time. Required value.`

`dataRetentionDateType`: The timestamp database column used to determine if a record is in the retention period. Defaults to `created_at`.

> `dataRetentionFilter`: Remove only records where the status of the record is "submitted" or "unsubmmited". Defaults to 'all' (both submitted and unsubmitted will be removed).

Reviewer: ```suggestion
`dataRetentionFilter`: Defines the subset of records to process based on their submission status. Remove only records where the status of the record is "submitted" or "unsubmmited". Defaults to "all" (both submitted and unsubmitted will be removed).
```
